### PR TITLE
Remove proxy settings detection behavior in "default proxy mode."

### PR DESCRIPTION
**Pull request by BillyONeal**
This commit partially reverts https://github.com/microsoft/cpprestsdk/commit/eb108ada1ab23a46a09efb87f35c802e57e0832a in order to work around a reliability problem with WinHttpGetProxyForUrl. That function hangs unless there is an available thread pool thread to complete the WPAD request. As a result, if a customer issued ~512 concurrent HTTP requests, or otherwise needed that many thread pool threads, there would not be a thread available for WinHTTP to complete the operation, and the program would deadlock.

Moreover this call to WinHttpGetDefaultProxyConfiguration is extremely expensive, taking ~20% of overall CPU for the entire program for some Azure Storage SDK customers.

The function WinHttpGetProxyForUrlEx is supposed to help with this problem by being asynchronous, but that function was added in Windows 8, so we can't use it unconditionally. And on Windows 8.1 we already are using WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY instead of trying to do proxy autodetect ourselves.

### CODE REVIEW

**Comment by BillyONeal** [comment]
@vadz Any comments?

**Comment by BillyONeal** [comment]
Ivan Pashov of the WinHTTP team confirms that WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY completely obsoletes WinHttpGetProxyForUrl.

**Comment by vadz** [comment]
@BillyONeal sorry for the delay with reply. In any case, I don't have much to say: I didn't notice any performance problems, but that's because I use the library from interactive application doing just a few requests. For me the important thing is that it works out of the box even in restricted corporate environments, but I think this should still be the case with your changes.

TL;DR: no real comments.

**Comment by BillyONeal** [comment]
@vadz In that case you will need to pass in a proxy config set to auto_discovery to request WPAD after this change..
